### ISSUE TITLE
Indentation is inconsistent by default between <tab> and the (normal mode) < and > commands.

### ISSUE THREAD

**Issue by etoombs-encircle** [mawww/kakoune#3136]
### Steps 
1. On the first line, enter a tab in insert mode with \<tab\> and write some non-whitespace text to ensure the tab stays.
1. One the second line, write some text, go back to normal mode, and indent once using >.

### Outcome 
The first line is indented with a single tab, which measures 8 spaces by default. The second line is indented using four spaces.

### Expected 
Both lines should have the same indentation and the same type of indentation. Either both in spaces or both in tabs.

Having \<tab\> enter tabs while doing the rest of the indentation with spaces and \<\> is not useful enough to keep, which is why every other editor maintains a link between indentation method and the tab key.

**Comment by Delapouite** [comment]
Hello.

Some explanations about this behavior in this thread https://github.com/mawww/kakoune/issues/3073

**Comment by etoombs-encircle** [comment]
> Hello.
> 
> Some explanations about this behavior in this thread #3073

Thanks, Delapouite. I figured that was the rationale here. Everything I said still applies. Assuming kak should be at its most useful by default, it should not limit `<tab>` to just tab character insertion. kak should use it as an insert-mode indentation command. All raw tab character insertion functionality remains in place. Only ^V is needed.

**Comment by etoombs-encircle** [comment]
Also, you clearly care about new user adoption. So, yeah.

**Comment by occivink** [comment]
> Doesn't work if the line is empty, but it's better.

You can use `<a-lt>` and `<a-gt>` instead of `<` and `>` in that case

**Comment by etoombs-encircle** [comment]
> > Doesn't work if the line is empty, but it's better.
> 
> You can use `<a-lt>` and `<a-gt>` instead of `<` and `>` in that case

Those buttons don't seem to do anything for me...

**Comment by etoombs-encircle** [comment]
> > Doesn't work if the line is empty, but it's better.
> 
> You can use `<a-lt>` and `<a-gt>` instead of `<` and `>` in that case

Thanks, @occivink . Editing the workaround now...

**Comment by occivink** [comment]
Yes, actually this should be fixed since a very recent commit, so try out master

**Comment by mawww** [comment]
I actually worked on that commit specifically due to this issue report, so that  we could reasonably point to this workaround.

**Comment by pickfire** [comment]
Even after than commit, tab still insert tab instead of spaces even with editorconfig turned on.
